zerver/views/realm.py

```diff
@@ -221,18 +222,25 @@ def update_realm(
         data["authentication_methods"] = authentication_methods
     # The message_editing settings are coupled to each other, and thus don't fit
     # into the do_set_realm_property framework.
+
+    message_content_edit_limit_seconds = realm.message_content_edit_limit_seconds
+    if message_content_edit_limit_seconds_raw is not None:
+        message_content_edit_limit_seconds = parse_message_content_edit_or_delete_limit(
+            message_content_edit_limit_seconds_raw,
+            Realm.MESSAGE_CONTENT_EDIT_OR_DELETE_LIMIT_SPECIAL_VALUES_MAP,
+            setting_name="message_content_edit_limit_seconds",
+        )
+
     if (
         (allow_message_editing is not None and realm.allow_message_editing != allow_message_editing)
         or (
-            message_content_edit_limit_seconds is not None
-            and realm.message_content_edit_limit_seconds != message_content_edit_limit_seconds
+            message_content_edit_limit_seconds_raw is not None
+            and message_content_edit_limit_seconds != realm.message_content_edit_limit_seconds
```

Comment:
I think this is not clean and we can improve this. We will make the topic editing permissions independent, so one condition will be removed then.

Also, I think we do not need to always update `allow_message_editing` and `message_content_edit_limit_seconds` together, since these are no longer controlled by single element in the UI. This settings are of course dependent, since the limit only matters if `allow_message_editing` is true, but I think we can remove this and handle it like other settings.

Also while sending events, we send all the settings in the object, even when those are not updated. It may makes sense when changing `allow_message_editing` from `False` to `True` but not in other cases. But I am not much sure about this.

Reply:
@timabbott what are your thoughts on this? Would be great to have feedback on this since working on further parts will depend on how we proceed with this.